src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -139,11 +145,14 @@ internal BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Dia
                                     checkedVariables,
                                     out conversion);
 
-            FailRemainingInferences(checkedVariables, diagnostics);
+            FailRemainingInferencesAndSetValEscape(checkedVariables, diagnostics, rightEscape);
 
-            var lhsTuple = DeconstructionVariablesAsTuple(left, checkedVariables, diagnostics, ignoreDiagnosticsFromTuple: hasErrors || !resultIsUsed);
+            var lhsTuple = DeconstructionVariablesAsTuple(left, checkedVariables, diagnostics, ignoreDiagnosticsFromTuple: diagnostics.HasAnyErrors() || !resultIsUsed);
             TypeSymbol returnType = hasErrors ? CreateErrorType() : lhsTuple.Type;
 
+            uint leftEscape = GetBroadestValEscape(lhsTuple, this.LocalScopeDepth);
+            boundRHS = ValidateEscape(boundRHS, leftEscape, isByRef: false, diagnostics: diagnostics);
```

Comment:
In the future we may want to do this pair-wise each value to each assignment target. 
For now, since in all reachable cases the assignment values will have the same escape which is equal to escape of RHS, this is sufficient.